api-tester/src/main/java/com/revenuecat/apitester/java/PurchasesAPI.java

```diff
@@ -0,0 +1,143 @@
+            @Override public void onReceived(@NonNull CustomerInfo customerInfo) {}
+            @Override public void onError(@NonNull PurchasesError error) {}
+        };
+        final LogInCallback logInCallback = new LogInCallback() {
+            @Override public void onReceived(@NotNull CustomerInfo customerInfo, boolean created) {}
+            @Override public void onError(@NotNull PurchasesError error) {}
+        };
+
+        purchases.syncPurchases();
+        purchases.getOfferings(receiveOfferingsListener);
+        purchases.getSubscriptionSkus(skus, skusResponseListener);
+        purchases.getNonSubscriptionSkus(skus, skusResponseListener);
+        purchases.purchaseProduct(activity, skuDetails, upgradeInfo, purchaseChangeListener);
+        purchases.purchaseProduct(activity, skuDetails, makePurchaseListener);
+        purchases.purchasePackage(activity, packageToPurchase, upgradeInfo, purchaseChangeListener);
+        purchases.purchasePackage(activity, packageToPurchase, makePurchaseListener);
+        purchases.restorePurchases(receiveCustomerInfoListener);
+
+        purchases.logIn("", logInCallback);
+        purchases.logOut();
+        purchases.logOut(receiveCustomerInfoListener);
+        final String appUserID = purchases.getAppUserID();
+        purchases.getCustomerInfo(receiveCustomerInfoListener);
+        purchases.removeUpdatedCustomerInfoListener();
+        purchases.invalidateCustomerInfoCache();
+        purchases.close();
+
+        final boolean finishTransactions = purchases.getFinishTransactions();
+        purchases.setFinishTransactions(true);
+        final UpdatedCustomerInfoListener updatedCustomerInfoListener = purchases.getUpdatedCustomerInfoListener();
+        purchases.setUpdatedCustomerInfoListener((CustomerInfo customerInfo) -> {});
+
+        final boolean anonymous = purchases.isAnonymous();
+
+        purchases.onAppBackgrounded();
+        purchases.onAppForegrounded();
+    }
+
+    static void check(final Purchases purchases, final Map<String, String> attributes) {
+        purchases.setAttributes(attributes);
+        purchases.setEmail("");
+        purchases.setPhoneNumber("");
+        purchases.setDisplayName("");
+        purchases.setPushToken("");
+        purchases.collectDeviceIdentifiers();
+        purchases.setAdjustID("");
+        purchases.setAppsflyerID("");
+        purchases.setFBAnonymousID("");
+        purchases.setMparticleID("");
+        purchases.setOnesignalID("");
+        purchases.setAirshipChannelID("");
+        purchases.setMediaSource("");
+        purchases.setCampaign("");
+        purchases.setAdGroup("");
+        purchases.setAd("");
+        purchases.setKeyword("");
+        purchases.setCreative("");
+    }
+
+    static void checkConfiguration(final Context context, final ExecutorService executorService) {
+        final List<? extends BillingFeature> features = new ArrayList<>();
```

Comment:
maybe we should have a TODO to add the builder version once amazon is merged?

Reply:
Done!